agent/provider/src/market/provider_market.rs

```diff
@@ -594,6 +594,38 @@ impl Handler<AgreementFinalized> for ProviderMarket {
                 error,
             );
         }
+
+        log::info!("Unsubscribing all active offers");
```

Comment:
Provider will unsubscribe and subscribe again all active Offers upon AgreementFinalized (in happy path it is after Activity is Destroyed and Invoice is sent to the Requestor; it happends also after Agreement expiration or in case of some internal provider errors).

Unsubscription and Subscription again is to enforce the Marketplace to emit fresh proposals.

```suggestion
        log::info!("Re-subscribing all active offers to get fresh proposals from the Market");
```